Makefile

```diff
@@ -12,57 +7,10 @@ include build-aux/help.mk
 include build-aux/k8s.mk
 include build-aux/teleproxy.mk
 
-.DEFAULT_GOAL = help
-
-export PATH:=$(CURDIR)/bin_$(GOOS)_$(GOARCH):$(PATH)
-
-test-cluster: $(KUBECONFIG) bin_$(GOOS)_$(GOARCH)/kubeapply
-	bin_$(GOOS)_$(GOARCH)/kubeapply -f k8s
-.PHONY: test-cluster
-
-test-suite.tap: .go-test.tap
-check: test-cluster
-go-test: test-cluster
-	$(MAKE) .go-test.tap.summary
-.go-test.tap: .go-test-nat.tap .go-test-teleproxy.tap .go-test-other.tap
-	@./build-aux/tap-driver cat $^ > $@
-# .tap-file                    [----firewall.lock----] [----cluster.lock----]               [---------------extra flags--------------] [----------------------------------------packages-----------------------------------------]
-.go-test-nat.tap      : FORCE; @$(FLOCK) .firewall.lock                        go test -json -exec sudo                                 $(go.module)/internal/pkg/nat                                                               | GO111MODULE=off go run build-aux/gotest2tap.go | tee $@ | build-aux/tap-driver stream -n $@
-.go-test-teleproxy.tap: FORCE; @$(FLOCK) .firewall.lock $(FLOCK) .cluster.lock go test -json -exec "sudo env KUBECONFIG=$${KUBECONFIG}" $(go.module)/cmd/teleproxy                                                                  | GO111MODULE=off go run build-aux/gotest2tap.go | tee $@ | build-aux/tap-driver stream -n $@
-.go-test-other.tap    : FORCE; @                        $(FLOCK) .cluster.lock go test -json                                            $$(go list ./... | grep -vF -e $(go.module)/internal/pkg/nat -e $(go.module)/cmd/teleproxy) | GO111MODULE=off go run build-aux/gotest2tap.go | tee $@ | build-aux/tap-driver stream -n $@
-
-ifneq ($(shell which docker 2>/dev/null),)
-.docker.tap: docker/teleproxy-check.docker
-	@$(if $(filter linux,$(GOOS)),$(FLOCK) .firewall.lock) docker run --rm --cap-add=NET_ADMIN "$$(sed -n 3p $<)" go test -json -exec sudo $(go.module)/internal/pkg/nat | GO111MODULE=off go run build-aux/gotest2tap.go | tee $@ | build-aux/tap-driver stream -n $@
-docker/teleproxy-check.docker: docker/teleproxy-check/teleproxy.tar
-docker/teleproxy-check/teleproxy.tar: go-get
-	rm -f $@
-	{ git ls-files; git ls-files --others --exclude-standard; } | while IFS='' read -r file; do \
-	    if [ -e "$$file" -o -L "$$file" ]; then \
-	        mkdir -p "$$(dirname "$(@D)/.tmp/teleproxy/$$file")"; \
-	        cp "$$file" "$(@D)/.tmp/teleproxy/$$file"; \
-	    fi; \
-	done; \
-	cd $(@D)/.tmp/teleproxy && go mod vendor
-	tar -c -f $@ -C $(@D)/.tmp/teleproxy .
-	rm -rf $(@D)/.tmp
-else
-.docker.tap: .go-test-nat.tap
-	@sed -En -e '/^TAP version/p' -e 's/^(not )?ok([^#]*)(#.*)?/ok\2 # SKIP/p' -e '/^[0-9]+\.\.[0-9]+$$/p' $< | tee $@ | build-aux/tap-driver stream -n $@
-endif
-test-suite.tap: .docker.tap
-
-clean:
-	$(FLOCK) .firewall.lock rm .firewall.lock
-	$(FLOCK) .cluster.lock rm .cluster.lock
-	rm -f -- .*.tap
+check: claim
```

Comment:
Maybe have this depend on `$(_kubernaut-ui.KUBECONFIG)` instead of `claim`?  In order to mesh with the hard-coded `const CLUSTER_FILE = "../../build-aux/cluster.knaut"` in the Go tests.

`make claim KUBECONFIG=other-file` will obey the `KUBECONFIG` variable and allow it to be overridden.

Reply:
done, although I think there is probably a bit more to resolve here as we go forward... note that the Manifests function allows you to override the location of kubeconfig via the TELEPROXY_DEV_KUBECONFIG environment variable